pkg/orchestrator/provider/azure/scanestimation/estimate.go

```diff
@@ -0,0 +1,262 @@
+// Copyright © 2023 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scanestimation
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v4"
+
+	"github.com/openclarity/vmclarity/api/models"
+	"github.com/openclarity/vmclarity/pkg/orchestrator/provider/common"
+	familiestypes "github.com/openclarity/vmclarity/pkg/shared/families/types"
+	"github.com/openclarity/vmclarity/pkg/shared/utils"
+)
+
+type MarketOption string
+
+const (
+	MarketOptionSpot     MarketOption = "Spot"
+	MarketOptionOnDemand MarketOption = "OnDemand"
+)
+
+type ScanEstimator struct {
+	priceFetcher PriceFetcherImpl
+}
+
+type EstimateAssetScanParams struct {
+	SourceRegion           string
+	DestRegion             string
+	DiskStorageAccountType armcompute.DiskStorageAccountTypes
+	StorageAccountType     armcompute.StorageAccountTypes
+	ScannerVMSize          armcompute.VirtualMachineSizeTypes
+	ScannerOSDiskSizeGB    int64
+	Stats                  models.AssetScanStats
+	Asset                  *models.Asset
+	AssetScanTemplate      *models.AssetScanTemplate
+}
+
+const halfAMinute = 30
+
+func New() *ScanEstimator {
+	netClient := &http.Client{
+		Timeout: time.Second * halfAMinute,
```

Comment:
Not sure what's the difference

Reply:
I think it reads better, but honestly I would prefer we move the whole timeout to a const instead of introducing "halfAMinute". like:
```
const scanEstimatorTimeout = time.Second * 30
```